tests/extraction_tests/test_round_three_extraction.py

```diff
@@ -314,6 +329,40 @@ def test_extract_funding_data(mock_funding_sheet, mock_project_lookup):
     assert_frame_equal(extracted_funding_data, expected_funding_data)
 
 
+def test_extract_funding_data_programme_only(mock_funding_sheet, mock_project_lookup):
+    """Test project level funding data extract does not include excluded data when programme only is selected"""
+    mock_funding_sheet.iloc[17, 4] = "Programme only"
+    extracted_funding_data = tf.extract_funding_data(mock_funding_sheet, mock_project_lookup)
+    assert not extracted_funding_data["Funding Source Name"].isin(["Town Deals 5% CDEL Pre-Payment"]).any()
+
+
+def test_extract_funding_data_fhsf(mock_funding_sheet, mock_project_lookup):
+    """Test project level funding data extract does not include excluded questions or excluded forcasts. when
+    Future_High_Street_Fund is selected"""
+    mock_project_lookup = {key: value.replace("TD", "HS") for (key, value) in mock_project_lookup.items()}
+    extracted_funding_data = tf.extract_funding_data(mock_funding_sheet, mock_project_lookup)
+    assert (
+        not extracted_funding_data["Funding Source Name"]
+        .isin(
+            [
+                "Town Deals 5% CDEL Pre-Payment",
+                "Towns Fund RDEL Payment which is being utilised on TF project related activity",
+                "How much of your RDEL forecast is contractually committed?",
+            ]
+        )
+        .any()
+    )
+    assert (
+        len(
+            extracted_funding_data[
+                (extracted_funding_data["Funding Source Type"] == "Towns Fund")
+                & (extracted_funding_data["Start_Date"] > datetime(2023, 10, 1))
```

Comment:
maybe could add a quick assertion to check that some rows are indeed extracted too, ie if start date is less than datetime(2023, 10, 1). Good to check the expected data as well as what has been removed

Reply:
Changed line 336 and 345 to assert on the expected length of each extracted dataframe 200, and 104 respectively